File: ios/Classes/FlutterStripeTerminalChannelHandler.swift
//
//  FlutterStripeTerminalChannelHandler.swift
//  flutter_stripe_terminal
//
//  Created by Vishal Dubey on 03/12/21.
//

import Flutter

public func handleMethodCall(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    guard let arguments = call.arguments as? [String : Any?] else {
        result(FlutterError(code: "NO_ARGUMENTS_PROVIDED", message: "No arguments were provided. If this call uses no arguments, provide an empty Map<String, Object>.", details: nil))
        return
    }
    
    switch(call.method) {
    case "setConnectionTokenParams":
        FlutterStripeTerminal.shared.setConnectionTokenParams(serverUrl: arguments["serverUrl"] as! String, authToken: arguments["authToken"] as! String, requestType: arguments["requestType"] as! String,  result: result)
    case "searchForReaders":
        FlutterStripeTerminal.shared.searchForReaders(simulated: arguments["simulated"] as! Bool,result: result)
    case "connectToReader":
        FlutterStripeTerminal.shared.connectToReader(readerSerialNumber: arguments["readerSerialNumber"] as! String, locationId: arguments["locationId"] as! String, result: result)
    case "processPayment":
        FlutterStripeTerminal.shared.processPayment(clientSecret: arguments["clientSecret"] as! String, result: result)
    case "updateReader":
        FlutterStripeTerminal.shared.updateReader(result: result)
    case "connectionStatus":
        FlutterStripeTerminal.shared.connectionsStatus(result: result)
    default:
        result(FlutterMethodNotImplemented)
    }
}

File: android/src/main/kotlin/in/agnostech/flutter_stripe_terminal/FlutterStripeTerminalChannelHandler.kt
package `in`.agnostech.flutter_stripe_terminal

import android.content.Context
import com.androidnetworking.AndroidNetworking
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class FlutterStripeTerminalChannelHandler(context: Context): MethodChannel.MethodCallHandler {

    init {
        AndroidNetworking.initialize(context)
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        when(call.method) {
            "setConnectionTokenParams" -> FlutterStripeTerminal.setConnectionTokenParams(call.argument<String?>("serverUrl")!!, call.argument<String?>("authToken")!!, result)
            "searchForReaders" -> FlutterStripeTerminal.searchForReaders(call.argument<Boolean>("simulated")!!, result)
            "connectToReader" -> FlutterStripeTerminal.connectToReader(call.argument<String>("readerSerialNumber")!!, call.argument<String>("locationId")!!, result)
            "processPayment" -> FlutterStripeTerminal.processPayment(call.argument<String>("clientSecret")!!, result)
            "connectionStatus" -> FlutterStripeTerminal.connectionStatus(result)
            "updateReader" -> FlutterStripeTerminal.updateReader(result)
            "disconnectReader" -> FlutterStripeTerminal.disconnectReader(result)
        }
    }

}